src/ansys/dpf/core/geometry.py

```diff
@@ -0,0 +1,403 @@
+"""Module containing the different geometry objects."""
+
+from ansys.dpf import core as dpf
+
+from ansys.dpf.core import Field
+from ansys.dpf.core.fields_factory import field_from_array
+from ansys.dpf.core.plotter import DpfPlotter
+
+import numpy as np
+
+
+def normalize_vector(vector):
+    """Normalize vector."""
+    return vector / np.linalg.norm(vector)
+
+
+class Points:
+    """
+    Collection of DPF points.
+
+    This can be a single point or a cloud of points.
+
+    Parameters
+    ----------
+    coordinates: array, list, Field
+        Coordinates of the points in a 3D space.
+
+    Examples
+    --------
+    Create Points object with two points, print object information and plot.
+
+    >>> from ansys.dpf.core.geometry import Points
+    >>> points = Points([[0, 0, 0], [1, 0, 0]])
+    >>> print(points)
+    DPF Points object:
+    Number of points: 2
+    Coordinates:
+      [[0. 0. 0.]
+      [1. 0. 0.]]
+    >>> points.plot()
+
+    """
+
+    def __init__(self, coordinates, server=None):
+        # Input check
+        if not isinstance(coordinates, Field):
+            coordinates = field_from_array(coordinates, server=server)
+
+        self._coordinates = coordinates
+        self._server = server
+
+    def __getitem__(self, value):
+        return self.coordinates.data[value]
+
+    def __len__(self):
+        return self.num_points
+
+    def __str__(self):
+        """Print Points information."""
+        txt = "DPF Points object:\n"
+        txt += f"Number of points: {self.num_points}\n"
+        txt += f"Coordinates:\n  {self._coordinates.data}\n"
+        return txt
+
+    @property
+    def coordinates(self):
+        """Coordinates of the Points."""
+        return self._coordinates
+
+    @property
+    def num_points(self):
+        """Total number of points."""
+        return (
+            self._coordinates.shape[0]
+            if isinstance(self._coordinates.shape, tuple)
+            else 1
+        )
+
+    @property
+    def dimension(self):
+        """Dimension of the Points object space."""
+        return 3
+
+    def plot(self, **kwargs):
+        """Visualize Points object."""
+        pl = DpfPlotter(**kwargs)
+        pl.add_points(self._coordinates.data)
+        pl.show_figure(show_axes=True)
```

Comment:
Does this accept to input our own plotter?? Should we return the plotter somehow?

Reply:
I saw the ``.plot()`` method of each oject as a quick way for the user to check that this is indeed the object they intended to creats. What would you like to do when returning the plotter?